### ISSUE TITLE
Allowing for future grants on tables and schemas

### ISSUE THREAD

**Issue by ian-r-rose** [Snowflake-Labs/roleout#3]
This is an exciting project to see take shape!

One limitation that I ran into is that it does not seem to be possible to add [future grants](https://community.snowflake.com/s/article/How-to-grant-select-on-all-future-tables-in-a-schema-and-database-level) to roles for schemas and tables within a database. Instead, we have to manually add all schema names and assign permissions to them.

In my particular deployment, we expect schemas to be fairly dynamic and be added regularly by analysts. So we want to have most permissioning to be happening at the database level. Terraform exposes this ability via the [`on_future`](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/schema_grant#on_future) parameter.

Would it be possible to add `on_future` support to schemas in a database for this project?

**Comment by sfc-gh-ckite** [comment]
@ian-r-rose to clarify, it sounds like you're requesting to be able to manage access (Read/ReadWrite/Full) at the database level? If so that is indeed a feature that is on the roadmap, but let me know if you're asking for something different.

**Comment by ian-r-rose** [comment]
> @ian-r-rose to clarify, it sounds like you're requesting to be able to manage access (Read/ReadWrite/Full) at the database level? If so that is indeed a feature that is on the roadmap, but let me know if you're asking for something different.

Yes, that's exactly right -- I was hoping to be able to set up some roles and grants before the schemas exist, and before we really know what they will be.
